src/jmclient/wallet_rpc.py

```diff
@@ -780,15 +781,24 @@ def directsend(self, request, walletname):
             # all error conditions).
             if not self.coinjoin_state == CJ_NOT_RUNNING:
                 raise ActionNotAllowed()
+
+            old_txfee = jm_single().config.get("POLICY", "tx_fees")
+            if "txfee" in payment_info_json:
+                jm_single().config.set("POLICY", "tx_fees",
+                                       str(payment_info_json["txfee"]))
+
             try:
                 tx = direct_send(self.services["wallet"],
                         int(payment_info_json["amount_sats"]),
                         int(payment_info_json["mixdepth"]),
                         destination=payment_info_json["destination"],
                         return_transaction=True, answeryes=True)
+                jm_single().config.set("POLICY", "tx_fees", old_txfee)
             except AssertionError:
+                jm_single().config.set("POLICY", "tx_fees", old_txfee)
```

Comment:
Maybe we should catch `ValueError` here too, since that's what's raised for an invalid int() cast?

Reply:
Don't think so, `jm_single().config.set()` will accept anything and, in any case, right place of sanity checks for configuration should be `src/jmclient/configure.py`.

Reply:
Yes, I see my error now, for some reason I was thinking about the `int()` casts but forgetting the txfee isn't included in the arguments, sorry.